src/datasets/arrow_dataset.py

```diff
@@ -198,27 +198,50 @@ class TensorflowDatasetMixin:
     _TF_DATASET_REFS = set()
 
     @staticmethod
-    def _get_output_signature(dataset, cols_to_retain, test_batch, batch_size):
+    def _get_output_signature(dataset: "Dataset", collate_fn: Callable, collate_fn_args: dict, batch_size: int):
+        """Private method used by `to_tf_dataset()` to find the shapes and dtypes of samples from this dataset
+           after being passed through the collate_fn.
```

Comment:
could we also have some docs on the output types ? (sorry for nitpicking ^^)

It can be useful if we ever want to improve this part of the code again

Reply:
Ah, sorry, sure!

Reply:
Done!